### PR TITLE
feat(graphql-model-transformer): set up transformer for sandbox mode directive

### PR DESCRIPTION
**Pull request by danielleadams**
#### Description of changes
Changes set up the transformers to take in the sandbox directive (`allow_public_data_access_with_api_key`) and write to the auth resolvers. When sandbox mode is enabled, it allows API Key auth; when disabled, calls without an `@auth` directive are unauthorized.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-core/src/graphql-api.ts [hunk 1]
```diff
@@ -198,7 +205,9 @@ export class GraphQLApi extends GraphqlApiBase implements GraphQLAPIProvider {
     this.schema = props.schema ?? new TransformerSchema();
     this.schemaResource = this.schema.bind(this);
 
-    if (props.createApiKey && modes.some(mode => mode.authorizationType === AuthorizationType.API_KEY)) {
+    const hasApiKey = modes.some(mode => mode.authorizationType === AuthorizationType.API_KEY);
```
Comment (SwaySway) [inline packages/amplify-graphql-transformer-core/src/graphql-api.ts:208]: does this check default and additional?

File: packages/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts [hunk 2]
```diff
@@ -442,3 +449,11 @@ async function _buildProject(opts: ProjectOptions<TransformerFactoryArgs>) {
   });
   return transform.transform(userProjectConfig.schema.toString());
 }
+
+async function _updateCurrentMeta(opts: ProjectOptions<TransformerFactoryArgs>) {
```
Comment (SwaySway) [inline packages/amplify-provider-awscloudformation/src/graphql-transformer/transform-graphql-schema.ts:453]: nit: this looks like a helper util we could store in utils
https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts

File: packages/amplify-graphql-transformer-core/src/graphql-api.ts (hunk 1 above)
Comment (danielleadams) [reply packages/amplify-graphql-transformer-core/src/graphql-api.ts:208]: Yep! https://github.com/aws-amplify/amplify-cli/blob/auth-dir-v-next/packages/amplify-graphql-transformer-core/src/graphql-api.ts#L176-L178

**Review by SwaySway — APPROVED** [review]
aside from nits lgtm

**Pull request merged**